### ISSUE TITLE
Run inside interpreter should change working directory

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#262]
_From [mproel...@googlemail.com](https://code.google.com/u/105494983286078857315/) on 2010-08-05T07:21:25Z_

Hello,

I face the following problem. I have a file let's say c:\test\test.py which uses a dll stored in the same path c:\test\test.dll

My code would be something like

import ctypes
ctypes.cdll.LoadLibrary("test.dll") #dll in same directory as test.py

when I "run and interact" everything works fine, but on "run inside interpreter" I get an error like the module (test.dll) was not found as the current working directory is not c:\test

Honestly I haven't got the difference between "run and interact" and "run inside interpreter" yet. What is the difference of these two commands and is the behaviour I have as expected?
(I keep searching the possibility to mark an issue as defect/enhancement)

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=262_

**Comment by spyder-issues-migrator** [comment]
_From [pierre.raybaut](https://code.google.com/u/pierre.raybaut/) on 2010-08-05T05:39:03Z_

(a simple user can't mark an issue as an enhancement which is unfortunate as it would be easier for me as well)

Well, actually you've just pointed out the difference between these two 'run' commands:
[1] "run and interact": open a brand new Python interpreter, run the script and interact with it right after its execution (this is strictly equivalent to 'python -i script.py' in a terminal, so the current working directory is the script directory)
[2] "run inside interpreter": run the script inside an already existing interpreter which had its own life before executing this script (namespace, working directory, environment variables, ...) -- as a consequence, results are expected to be different from [1]. This is almost equivalent to the following actions performed in a terminal:  open a Python interpreter, do what you want inside of it (computings, ...) and run your script with the 'execfile' function.

In other words, [1] is to be used when you want to run your script as you would outside Spyder or any other IDE but you want to interact with it (for debugging purpose for example). [2] is to be used when you want to run one or more script(s) inside an existing interpreter in which you have already performed some computings and with data stored in the namespace that you don't want to loose between each script execution.

**Status:** WontFix
